tests/functional/metrics/test_metrics_provider.py

```diff
@@ -0,0 +1,182 @@
+import json
+from collections import namedtuple
+from typing import Any, List
+
+import pytest
+
+from aws_lambda_powertools.metrics import (
+    EphemeralMetrics,
+    Metrics,
+    SchemaValidationError,
+)
+from aws_lambda_powertools.metrics.provider import BaseProvider
+from aws_lambda_powertools.metrics.provider.base import reset_cold_start_flag_provider
+
+
+def capture_metrics_output(capsys):
+    return json.loads(capsys.readouterr().out.strip())
+
+
+@pytest.fixture
+def metrics_provider() -> BaseProvider:
```

Comment:
is that correct? looks like a leftover from the Protocol implementation. Now that we have an ABC, it'd fail without `clear_metrics`, but it doesn't because we're implementing a standalone class.